backend/event/mutations/UpdateEpisodeOrderMutation.py

```diff
@@ -15,17 +15,30 @@ class Arguments:
 
     @classmethod
     def mutate(cls, root: None, info: ResolveInfo, episode_ids: list[str]):
-        source = Source.objects.filter(episode__in=episode_ids).distinct()
+        if len(episode_ids) == 0:
+            error=LettercraftErrorType(
+                field="episode_ids",
+                messages=["No episode IDs provided."]
+            )
+            return cls(ok=False, errors=[error]) # type: ignore
 
-        if source.count() > 1:
+        corresponding_sources = Source.objects.filter(episode__in=episode_ids)
+        if corresponding_sources.count() != len(episode_ids):
             error = LettercraftErrorType(
                 field="episode_ids",
-                messages=["Multiple sources found for given episode IDs."],
+                messages=["Not every episode has a corresponding source."]
             )
-            return cls(ok=False, errors=[error])  # type: ignore
+            return cls(ok=False, errors=[error]) # type: ignore
```

Comment:
This check is hard to understand without a comment since it relies on Django returning duplicate rows for the `filter` query, which is kind of obscure knowledge. It's also an indirect test for the thing being validated (invalid IDs), which also makes it less intuitive.

The message here is also confusing. Strictly speaking, every episode *does* have a corresponding source, but not every ID has a corresponding episode. Practically, the feedback to the client should be that it's sending non-existent episode IDs, not that there is an issue in the `source` field of episodes.

Reply:
Wow, so basically everything here is bad then 😅

I've made some changes in my last commit, but if this is still not how you want it to be, feel free to push your own commits.

Reply:
I mean, it was doing the right thing, just confusing to read. Your updated version looks fine!